fpa/fastapi/fastapi_add.go

```diff
@@ -110,6 +131,15 @@ func (i *Instance) storeInFIL(ctx context.Context, c cid.Cid) (ColdInfo, error)
 	return ci, nil
 }
 
+func (i *Instance) addToHotLayer(ctx context.Context, reader io.Reader) (*cid.Cid, error) {
+	path, err := i.ipfs.Unixfs().Add(ctx, files.NewReaderFile(reader), options.Unixfs.Pin(false))
```

Comment:
You guys might consider trying to combine the pinning and adding. I'm guessing the same pin will be added in a different step.